serveradmin/serverdb/models.py

```diff
@@ -324,7 +324,10 @@ def get_supernet(self, servertype):
         )
 
     def clean(self, *args, **kwargs):
-        super(Server, self).clean(*args, **kwargs)
+        # The super method does actually nothing but we keep this here in
+        # case this changes.
+        super(Server, self).clean()
```

Comment:
>  So, I suggest to add *args and **kwargs again

Yeah this is probably safer. I removed it because the code style check made an complaint about passung unused paramters.

> Also, this comment is not very useful in here.

When thinking again about it I It was more a note/reminder to myself. I will remove it.

> If super().clean() didn't need calling, then there wouldn't be such a function.

I would like to agree but from what I can see in this particular case the function is really just present for design or interface description. From the super clean method:

```
   # django.db.models.base

    def clean(self):
        """
        Hook for doing any extra model-wide validation after clean() has been
        called on every field by self.clean_fields. Any ValidationError raised
        by this method will not be associated with a particular field; it will
        have a special-case association with the field defined by NON_FIELD_ERRORS.
        """
        pass
```

However I agree that we should call it and it makes sense to pass it args and kwargs.

Reply:
Looking at the Sourcecode of the base method and the documentation: https://docs.djangoproject.com/en/2.2/ref/models/instances/#django.db.models.Model.clean again I think this method should not have parameters at all. As @TheFRedFox pointed out in DM args and kwargs are not used in the methods body and I could not find any usages of it with parameters so I think it is safe to remote the parameters.

Reply:
Here, as django is not using the abstraction library (https://docs.python.org/3/library/abc.html) I see this method being meant as an abstract method, meant to be implemented in a concrete model class.
Therefore, not calling the super-method is ok in my eyes, however to follow the inheritance and calling the super-method even though it is meant as being an abstract one, is fine, too.

As the "abstract" clean is not taking any arguments and we are not using any arguments, I would also say to actually get rid of the extra parameters in our implementation.